### ISSUE TITLE
Support declarationMap

### ISSUE THREAD

**Issue by mrjackdavis** [privatenumber/pkgroll#81]
### Feature request

pkgroll bundles don't include sourcemaps for declaration files

See https://www.typescriptlang.org/tsconfig/#declarationMap

### Motivations

I want to implement pkgroll across all packages in our monorepo, but we need source "click-through" capability which depends on declarationMap

### Bugs are expected to be fixed by those affected by it

- [ ] I'm interested in working on this issue

### Compensating engineering work financially will speed up resolution

- [X] I'm willing to offer financial support

**Comment by privatenumber** [comment]
Interesting, I didn't even know this was a thing.

We use https://github.com/Swatinem/rollup-plugin-dts to bundle declaration files.

If anyone wants to help, we'll have to investigate whether this supports source maps first or if we should explore alternatives.

**Comment by mrjackdavis** [comment]
Yeah it's not often mentioned but crucial to a good monorepo experience

After posting this issue I did some digging. Looks like that rollup plugin is in maintenance mode and doesn't support this feature.

It therefore is likely cumbersome to implement this feature for pkgroll. I spent some time today making our own, very simple, pkgroll equivalent using esbuild and calling tsc with declaration only. It will tide us over for now
